matrix/matrix.lua

```
@@ -0,0 +1,210 @@
+		asn_org = "AS Organization unknown"
+	else
+		asn_org = tostring(asn_org)
+	end
+	local data = {}
+	data["formatted_body"] = "<p>Denied " .. request_method .. " from <b>" .. remote_addr .. "</b> (" .. country .. " • \"<i>" .. asn_org .. "</i>\" • " .. asn .. ") to " .. request_host .. self.ctx.bw.uri .. "<br>"
+	data["formatted_body"] = data["formatted_body"] .. "Reason <b>" .. reason .. "</b> (" .. encode(reason_data or {}) .. ").</p>"
+	data["body"] = "Denied " .. request_method .. " from " .. remote_addr .. " (" .. country .. " • \"" .. asn_org .. "\" • " .. asn .. ") to " .. request_host .. self.ctx.bw.uri .. "\n"
+	data["body"] = data["body"] .. "Reason " .. reason .. " (" .. encode(reason_data or {}) .. ")."
+	-- Add headers if enabled
+	if self.variables["MATRIX_INCLUDE_HEADERS"] == "yes" then
+		local headers, err = ngx_req.get_headers()
```

Luacheck (1.2.0)

[warning] shadowing definition of variable 'err' on line 62

(W421)